api/admin/controller/cdn_services.py

```diff
@@ -85,15 +85,20 @@ def process_post(self):
 
     def validate_form_fields(self, **fields):
         """The 'name' and 'protocol' fields cannot be blank, and the protocol must
-        be selected from the list of recognized protocols."""
+        be selected from the list of recognized protocols.  The URL must be valid."""
 
         name = fields.get("name")
         protocol = fields.get("protocol")
 
         if not name:
             return INCOMPLETE_CONFIGURATION
-        if protocol and protocol not in [p.get("name") for p in self.protocols]:
-            return UNKNOWN_PROTOCOL
+        if protocol:
+            if protocol not in [p.get("name") for p in self.protocols]:
+                return UNKNOWN_PROTOCOL
```

Comment:
This little bit could also move into validate_formats in the near future -- you use it a lot.